src/Datadog.Trace.ClrProfiler.Native/cor_profiler.cpp

```diff
@@ -582,6 +591,88 @@ HRESULT STDMETHODCALLTYPE CorProfiler::JITCompilationStarted(
   return S_OK;
 }
 
+//
+// ICorProfilerCallback6 methods
+//
+HRESULT STDMETHODCALLTYPE CorProfiler::GetAssemblyReferences(
+    const WCHAR* wszAssemblyPath,
+    ICorProfilerAssemblyReferenceProvider* pAsmRefProvider) {
+  auto assemblyPathString = ToString(wszAssemblyPath);
+  auto filename =
+      assemblyPathString.substr(assemblyPathString.find_last_of("\\/") + 1);
+  auto lastNiDllPeriodIndex = filename.rfind(".ni.dll");
+  auto lastDllPeriodIndex = filename.rfind(".dll");
+  if (lastNiDllPeriodIndex != std::string::npos) {
+    filename.erase(lastNiDllPeriodIndex, 7);
+  } else if (lastDllPeriodIndex != std::string::npos) {
+    filename.erase(lastDllPeriodIndex, 4);
+  }
+
+  // Get assembly name
+  const WSTRING assembly_name = ToWSTRING(filename);
+
+  // We must never try to add assembly references to
+  // mscorlib or netstandard. Skip other known assemblies.
+  for (auto&& skip_assembly : skip_assemblies) {
+    if (assembly_name == skip_assembly) {
+      Debug("GetAssemblyReferences skipping known assembly: ", wszAssemblyPath);
+      return S_OK;
+    }
+  }
+
+  std::vector<IntegrationMethod> filtered_integrations =
+      FlattenIntegrations(integrations_);
+
+  // TODO: Make this assembly reference dynamic vs hard-coded
+  const AssemblyReference assemblyReference = trace::AssemblyReference(
+      L"Datadog.Trace.ClrProfiler.Managed, Version=1.16.2.0, Culture="
+      L"neutral, PublicKeyToken=def86d061d0d2eeb");
+
+  ASSEMBLYMETADATA assembly_metadata{};
+
+  assembly_metadata.usMajorVersion = assemblyReference.version.major;
+  assembly_metadata.usMinorVersion = assemblyReference.version.minor;
+  assembly_metadata.usBuildNumber = assemblyReference.version.build;
+  assembly_metadata.usRevisionNumber = assemblyReference.version.revision;
+  if (assemblyReference.locale == "neutral"_W) {
+    wchar_t nulltermEmptyStr[2] = L"\0";
```

Comment:
I'm sure any value >1 works here since it's null-terminated, but isn't the length of this array only 1?

Reply:
Outdated so we can resolve this. But the IDE helpers did show that `L"\0"` did require a size of 2. Not sure why.

Reply:
Oh 🤦 
The compiler adds a null terminator after string literals, so `"\0"` ends up becoming 2 null terminators. For an empty string, this would have worked:
```cpp
 wchar_t nulltermEmptyStr[1] = L"";

// nulltermEmptyStr[0] == '\0'
```